fem/tmop.hpp

```diff
@@ -676,7 +706,12 @@ class DiscreteAdaptTC : public TargetConstructor
 protected:
```

Comment:
Can you `#ifdef MFEM_USE_MPI` the forward definition of `class ParGridFunction` above?

Without MPI none of the Par objects or classes should be visible...